### ISSUE TITLE
[ORCHID]

### ISSUE THREAD

**Issue by darkk99** [Vanilla-OS/live-iso#279]
### Issue Description

Post-install setup on Orchid requires you to boot into previous partition rather than current, otherwise hostname is wrong and you can't sign in to TTY nor GDM.

### Steps to Reproduce

- Install VanillaOS Orchid
- Do post-setup
- Boot into "current partition"
- Try signing into GDM
- Go to TTY to check hostname and try signin

### On what version of Vanilla OS this happens?

Unreleased

### Additional Information

This was done using a manual install on `nvme0n1` where `nvme0n1p1` is a 465GiB NTFS partition.

**Comment by taukakao** [comment]
Booting into the past patition is not a solution herw since it misses everything that was installed by first-setup. 

Does rebooting into the current partition help?

**Comment by darkk99** [comment]
UPDATE: Looks like you *always* have to boot into option A on grub, *not* just the first boot after post-install

**Comment by taukakao** [comment]
No @darkk99 after first install you have to boot into B.

**Comment by darkk99** [comment]
> Booting into the past patition is not a solution herw since it misses everything that was installed by first-setup.
> 
> Does rebooting into the current partition help?

It doesn't seem to mess *anything* up. Post-install doesn't reappear or anything, all selected software is installed, updates work (check my last comment)

**Comment by darkk99** [comment]
> No @darkk99 after first install you have to boot into B.

This causes you to be stuck in GDM on latest (93) ISO

**Comment by taukakao** [comment]
Also @MonsterObserver I don't think that's whats happening here.
I noticed there were some issues with the host name. NetworkManager complained to me about something a while ago.

**Comment by MonsterObserver** [comment]
> Also @MonsterObserver I don't think that's whats happening here. I noticed there were some issues with the host name. NetworkManager complained to me about something a while ago.

Yeah not sure, in any case if the other root works fine, this is my best guess.

**Comment by taukakao** [comment]
> It doesn't seem to mess _anything_ up. Post-install doesn't reappear or anything, all selected software is installed, updates work (check my last comment)

That's very weird. Flatpak should not be installed on A if this is after the reboot after first install?

**Comment by taukakao** [comment]
Did you boot into A for first setup by any chance?

**Comment by darkk99** [comment]
> Did you boot into B for first setup by any chance?

No, booted into A.

**Comment by taukakao** [comment]
Hm, I guess we'll have to see if this is a reaccuring issue. I can really not make sense of this situation.

**Comment by taukakao** [comment]
Oh, and to fix your second root you can run `abroot pkg apply` in Root A and reboot.

**Comment by taukakao** [comment]
I can't reproduce this issue with # 93

**Comment by VladShmi** [comment]
This also keeps happening to me, and console is not oppening.

**Comment by taukakao** [comment]
@VladShmi 
The console is broken at the moment, this is being worked on. As a workaround you can:
1. Press Alt+F2
2. Type in `vso pico-init`, press enter and wait for about a minute
3. Open Console again

**Comment by nisel11** [comment]
I have the same problem when installing on my laptop, hopefully this will be fixed during Beta

**Comment by taukakao** [comment]
@nisel11 
Since I can not reproduce this issue I depend on additional information. 
So without it it can't be fixed. 
Valuable information would be: is anything special about your installation, how did you create the installation media, are you noticing anything else out of the ordinary, are there any warning messages or logs that might be interesting here and so on.

**Comment by nisel11** [comment]
There is nothing in vos-b that was in vos-a, not even flatpak applications, it's like two different systems, only one works fine and the other just exists.

**Comment by nisel11** [comment]
After abroot upgrade --force everything is back to what it was before all my actions, but now if I create a user I can't log into gnome, it just hangs on a gray screen and restarts gdm, maybe it's because I missed the initial setup this time. This all seems very strange, I guess hoping it will be fixed is not worth it

**Comment by taukakao** [comment]
@nisel11 
Did you use automatic partitioning or did you manually partition it?

**Comment by taukakao** [comment]
Also, the output of host-spawn lsblk or if that doesn't work just lsblk inside the broken root would be quite interesting.

**Comment by nisel11** [comment]
Manually, I like Linux, but I also need Windows on the second partition

**Comment by nisel11** [comment]
![2024-01-10-09-59-00-526.jpg](https://github.com/Vanilla-OS/live-iso/assets/96194200/79b4b874-c7b9-427a-bf25-f6ccd8b910b9)

in vos-a

![2024-01-10-10-01-59-515.jpg](https://github.com/Vanilla-OS/live-iso/assets/96194200/cd866be0-2046-4d64-9424-0b788825a8b1)

in vos-b

**Comment by nisel11** [comment]
![2024-01-10-10-05-15-453.jpg](https://github.com/Vanilla-OS/live-iso/assets/96194200/70d230d2-636b-47de-a392-3f53bf117755)

**Comment by taukakao** [comment]
@nisel11 
That looks very wrong. 
The home directory and the root should absolutely not be on the same partition. 

Do you still know by any chance how you did the manual partitioning and/or what the other partitions on your drive are?

**Comment by nisel11** [comment]
The manual partitioning was done with the help of a previous developer blog. I also logged into vos-b via grub and saw the following
![2024-01-10-10-16-46-453.jpg](https://github.com/Vanilla-OS/live-iso/assets/96194200/a0293cde-44c5-4e67-9b4f-3659423ccb98)

**Comment by taukakao** [comment]
@nisel11 
Yeah, you're completely missing the var partition, which would usually hold all variable data like your home folder or flatpak apps. 

I will do some testing to see if I can reproduce this issue.

It would be really useful to know which guide you used, if you can still find it.

**Comment by nisel11** [comment]
nvme0n1p1 - windows boot manager
nvme0n1p2 - some space also reserved by windows
nvme0n1p3 - windows
nvme0n1p4 - boot partition VanillaOS 1024 mb
nvme0n1p5 - efi partition VanillaOS 512 mb
nvme0n1p6 - root partition with 20992 mb
nvme0n1p7 - var partition

I just allocated 40GB from windows and created partitions via gparted using this guide![Screenshot_20240110-103001.png](https://github.com/Vanilla-OS/live-iso/assets/96194200/b72aa45c-bcd0-4811-94e4-32b97c330f89)

**Comment by taukakao** [comment]
@nisel11 
Thanks, the issue is that for some reason your var partition is used as root a. 

I will look into it, thanks for your input.

**Comment by nisel11** [comment]
I was able to see this problem on a virtual machine! You just need to use manual partitioning, after the first setup you just can't get into vos-b, after abroot upgrade --force you will get the problem this question started with. I did all of this on VMware Workstation 17

**Comment by nisel11** [comment]
Nothing has changed in build 101

**Comment by taukakao** [comment]
@nisel11 
Whoops, seems like it was me who broke ABRoot a while back. 
Sorry for the inconvenience ^^'
But I found the bug and it's going to be fixed soon.
Thanks for your help
